internal/provider/requestable_permission_data_source.go

```diff
@@ -61,72 +58,33 @@ func (d *requestablePermissionDataSource) Configure(ctx context.Context, req dat
 		return
 	}
 
-	apiToken, ok := req.ProviderData.(string)
+	client, ok := req.ProviderData.(*LumosAPIClient)
 	if !ok {
 		resp.Diagnostics.AddError(
 			"Unexpected Data Source Configure Type",
-			fmt.Sprintf("Expected string, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+			fmt.Sprintf("Expected *LumosAPIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
 		)
 
 		return
 	}
 
-	d.apiToken = apiToken
+	d.client = client
 }
 
 func (d *requestablePermissionDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
 	var state requestablePermissionDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
```

Comment:
Do we need to remove this?

Reply:
if we want to pass permissions by name instead, this can stay, right?